lib/eventsource.js

```diff
@@ -324,12 +348,14 @@ function EventSource (url, eventSourceInitDict) {
     }
 
     req.on('error', function (err) {
-      failed({ message: err.message })
+      failOnce({ message: err.message })
     })
 
     req.on('timeout', function () {
-      failed({ message: 'Read timeout, received no data in ' + config.readTimeoutMillis +
+      failOnce({ message: 'Read timeout, received no data in ' + config.readTimeoutMillis +
           'ms, assuming connection is dead' })
+      // Timeout doesn't mean that the request is cancelled, just that it has elapsed the timeout.
+      destroyRequest()
```

Comment:
Being more careful about destroying requests will help to ensure applications are not held open by the event source.